File: 70-climbing-stairs/climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        # climbStairs(n) = climbStairs(n-1) + climbStairs(n-2)
        # if n == 1:
        #     return 1
        # if n == 2:
        #     return 2
        # return self.climbStairs(n-1) + self.climbStairs(n-2)
        res = [0, 1, 2]
        for i in range(3, n+1):
            res.append(res[i-1] + res[i-2])
        return res[n]

File: 70-climbing-stairs/climbing-stairs.ts
const steps = [1, 1, 2];

function climbStairs(n: number): number {
    if (steps[n]) {
        return steps[n];
    }
    steps[n] = climbStairs(n-1) + climbStairs(n-2);
    return steps[n];
};